ros2_batch_job/__main__.py

```diff
@@ -382,21 +413,9 @@ def build_and_test(args, job):
     # xunit2 format is needed to make Jenkins xunit plugin 2.x happy
     with open('pytest.ini', 'w') as ini_file:
         ini_file.write('[pytest]\njunit_family=xunit2')
-    # check if packages have a pytest.ini file and add the xunit2
-    # format if it is not present
-    for path in Path('.').rglob('pytest.ini'):
-        config = configparser.ConfigParser()
-        config.read(str(path))
-        try:
-            # only if xunit2 is set continue the loop with the file unpatched
-            if config.get('pytest', 'junit_family') == 'xunit2':
-                continue
-        except configparser.NoOptionError:
-            pass
-        print('xunit2 patch applied to ' + str(path))
-        config.set('pytest', 'junit_family', 'xunit2')
-        with open(path, 'w+') as configfile:
-            config.write(configfile)
+    # check if packages have a pytest.ini file that will override xunit2 setup
+    # and patch configuration if needed to keep the xunit2 configuration
+    force_xunit2_in_pytest_ini_files()
```

Comment:
Why move the code into a function when it is only used in this single location? It makes the diff much more difficult to read.

Reply:
It tried to reduce the number of code lines in the main code path. Changed in 4e70a79